### PR TITLE
perf: use hot worker threads when possible

### PR DESCRIPTION
**Pull request by AlliBalliBaba**
This PR is a first small step towards auto-scaling worker threads. The main idea is to re-use 'hot threads'.

Workers are long running processes, they often persist objects or external connections across multiple requests. Dispatching requests Round-Robin style will sometimes cause the opening and closing of unnecessary connections.

This is easily optimized by dispatching to all threads in a specified order. This way we are mainly using a number of 'hot' worker threads for requests, while 'cold' threads can stay idle for potential latency spikes.

While it seems like this approach will cause additional overhead, it actually reduces overhead slightly (especially on a high number of threads).

### Benchmarks

`wrk -t4 -c150 -d60 http://localhost` (done locally)

`20 CPU cores 40 threads 'Hello world'`
main branch: `~72000 RPS`
this branch:     `~75000 RPS`

`20 CPU cores 100 threads 'IO simuation'`
main branch: `~28000 RPS`
this branch:     `~33000 RPS`

The 'IO simulation' does this in addition to the 'Hello world':
```
// do 10_000 assignments
for($i=0; $i<10_000; $i++) {
    $a = $i;
}
// sleep 1 millisecond
usleep(1000); 
```

### CODE REVIEW

**Comment by AlliBalliBaba** [comment]
Not sure if this would still be called 'Round Robin'. We could also keep it as a 'dispatch_mode' configuration, like [Swoole](https://openswoole.com/docs/modules/swoole-server/configuration#dispatch_mode) does. I'm not sure though if regular Round Robin would ever be preferred.

**Comment by dunglas** [comment]
Let's keep only the best mode, unless there is a use case for others.

File: worker.go [hunk 1]
```diff
@@ -242,7 +262,10 @@ func assignThreadToWorker(thread *phpThread) {
 	if !workersAreReady.Load() {
 		workersReadyWG.Done()
 	}
-	// TODO: we can also store all threads assigned to the worker if needed
+	thread.requestChan = make(chan *http.Request)
+	worker.threadMutex.Lock()
+	defer worker.threadMutex.Unlock()
```
Comment (dunglas) [inline worker.go:267]: `defer` doesn't seem very useful here and may cost one extra allocation. Maybe can we just must the call to `Unlock()` after the call to `append()`?

**Review by dunglas — APPROVED** [review]
Good idea!

File: worker.go (hunk 1 above)
Comment (AlliBalliBaba) [reply worker.go]: Interesting, I didn't know defer allocates. Changed it 👍

**Comment by AlliBalliBaba** [comment]
Was hoping for my actions to back this month 😞 , guess it's a bug then

**Comment by dunglas** [comment]
Thank you!!

**Pull request merged**